step-controller/step-controller-server/src/main/java/step/controller/services/entities/AbstractEntityServices.java

```diff
@@ -197,4 +198,36 @@ public History(String id, long updateTime) {
     public T restoreVersion(@PathParam("id") String id, @PathParam("versionId") String versionId) {
         return accessor.restoreVersion(new ObjectId(id), new ObjectId(versionId));
     }
+
+    @Operation(operationId = "is{Entity}Locked", description = "Get entity locking state")
+    @GET
+    @Path("/{id}/locked")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Secured(right = "{entity}-read")
+    public boolean isLocked(@PathParam("id") String id) {
+        T t = getEntity(id);
+        Boolean locked = t.getCustomField(CUSTOM_FIELD_LOCKED, Boolean.class);
+        return (locked != null && locked);
+    }
+
+    @Operation(operationId = "lock{Entity}", description = "Lock this entity")
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
```

Comment:
Not sure if the boolean payload matches with this. This might create errors in client generators

Reply:
this is valid for boolean and work with the FE client genetor